File: Week2/Code/boilerplate.py
#!/usr/bin/env python3 

#Author: Emma Deeks ead19@imperial.ac.uk
#Script: boilerplate.py
#Desc:  Example exercise on writing python programmes. Programme prints 'This is a boilerplate'). 
# Also illustrates use of docstrings
#Arguments: No input
#Outputs: 'This is a boilerplate'
#Date: Oct 2019 

"""This script illustrates how to writing python programmes using main arguments to import functions to modules """

__appname__ = '[boilerplate]'
__author__ = 'Emma Deeks (ead19@ic.ac.uk)'
__version__ = '0.0.1'
__licence__ = "License for this code/program"

## imports ##
import sys # Module to interface our program with the operating system 

## functions ##
def main(argv):  # this is defining a function 
    """ Main entry point of the program """ #Only appears in troubleshooting 
    print('This is a boilerplate') # NOTE: indented using two tabs or 4 spaces
    return 0
# Makes file usable as a script as well as a importable module 
if __name__ == "__main__":   # Turning into a programme- executable externally 
    """Makes sure the "main" function is called from command line"""  
    status = main(sys.argv)
    sys.exit(status)

#Import this module in python or ipython shell 

File: week3/code/boilerplate.R
#!/usr/bin/env Rscript 

#Author: Emma Deeks ead19@imperial.ac.uk
#Script: boilerplate.R
#Desc: Illustrates how R functions work
#Arguments: Two arguments defined within script	
#Outputs: Class of inputs
#Date: Oct 2019  


#Illustrates how R functions work and takes as input 
#Already defined inputs and returns the class of these inputs

MyFunction <- function(Arg1, Arg2){

  # Statements involving Arg1, Arg2:
  print(paste("Argument", as.character(Arg1), "is a", class(Arg1))) # print Arg1's type
  print(paste("Argument", as.character(Arg2), "is a", class(Arg2))) # print Arg2's type

  return (c(Arg1, Arg2)) #this is optional, but very useful
}

MyFunction(1,2) #test the function
MyFunction("Riki","Tiki") #A different test
